src/core/Akka/Dispatch/MessageQueues/UnboundedDequeMessageQueue.cs

```diff
@@ -19,6 +21,8 @@ public BoundedDequeMessageQueue()
             : base(new BoundedMessageQueue())
         {
         }
+
+        public TimeSpan PushTimeOut { get; set; }
```

Comment:
Maybe this needs a new base class that understands bounded semantics and thus can use the PushTimeout.
(?)